File: Controlador/CajaControlador.py
from Modelo.Movimiento import Movimiento, Tipo as TipoMovimiento
from datetime import datetime

class CajaControlador:
    def __init__(self, app):
        self.app = app
        self.panel = None

    def setPanel(self, panel):
        self.panel = panel

    def abrirCaja(self, fondo):
        self.app.abrirCaja(fondo)
        if self.panel is not None:
            self.panel.refrescar()

    def registrarRetiro(self, concepto, monto):
        self.app.registrarRetiro(concepto, monto)
        if self.panel is not None:
            self.panel.refrescar()

    def registrarIngresoExtra(self, concepto, monto):
        m = Movimiento()
        m.setTipo(TipoMovimiento.VENTA)
        m.setDescripcion(concepto)
        m.setMonto(monto)
        m.setFecha(datetime.now())
        m.setUsuario(self.app.getUsuarioActivo().getNombre() if self.app.getUsuarioActivo() else "SISTEMA")
        self.app.getMovimientoBD().insertar(m)
        self.app.registrarVentaSimple(monto)
        if self.panel is not None:
            self.panel.refrescar()

    def getTotalVentas(self):
        return sum(m.getMonto() for m in self.app.getMovimientos() if m.getTipo() == TipoMovimiento.VENTA)

    def getTotalEgresos(self):
        return sum(m.getMonto() for m in self.app.getMovimientos() if m.getTipo() == TipoMovimiento.RETIRO)

    def getEfectivoEsperado(self):
        return self.app.getMontoCaja()

    def getMovimientos(self):
        return self.app.getMovimientos()

    def isCajaAbierta(self):
        return self.app.isCajaAbierta()

    def getMontoCaja(self):
        return self.app.getMontoCaja()

File: Controlador/CajaControlador.cs
using System;
using System.Collections.Generic;
using System.Linq;
using punto_de_venta_C_.Modelo;
using punto_de_venta_C_.Vista;

namespace punto_de_venta_C_.Controlador
{
    public class CajaControlador
    {
        private readonly ControladorPrincipal app;
        private CajaPanel panel;

        public CajaControlador(ControladorPrincipal app)
        {
            this.app = app;
        }

        public void setPanel(CajaPanel panel)
        {
            this.panel = panel;
        }

        public void abrirCaja(double fondo)
        {
            app.abrirCaja(fondo);
            if (panel != null)
                panel.refrescar();
        }

        public void registrarRetiro(string concepto, double monto)
        {
            app.registrarRetiro(concepto, monto);
            if (panel != null)
                panel.refrescar();
        }

        public void registrarIngresoExtra(string concepto, double monto)
        {
            app.getMovimientoBD().insertar(new Movimiento(Movimiento.Tipo.VENTA, concepto, monto, DateTime.Now,
                    app.getUsuarioActivo().getNombre()));
            app.registrarVentaSimple(monto);
            if (panel != null)
                panel.refrescar();
        }

        public double getTotalVentas()
        {
            return app.getMovimientos().Where(m => m.getTipo() == Movimiento.Tipo.VENTA)
                    .Select(m => m.getMonto()).Sum();
        }

        public double getTotalEgresos()
        {
            return app.getMovimientos().Where(m => m.getTipo() == Movimiento.Tipo.RETIRO)
                    .Select(m => m.getMonto()).Sum();
        }

        public double getEfectivoEsperado()
        {
            return app.getMontoCaja();
        }

        public List<Movimiento> getMovimientos()
        {
            return app.getMovimientos();
        }

        public bool isCajaAbierta()
        {
            return app.isCajaAbierta();
        }

        public double getMontoCaja()
        {
            return app.getMontoCaja();
        }
    }
}
